File: CS/WizardCustomizationNamePage/DataSourceWizardCustomization.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardCommon.DataSourceWizard;
using DevExpress.DashboardWin.ServiceModel;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.DataAccess.Wizard.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WizardCustomizationExample {
    public class DataSourceWizardCustomization : IDashboardDataSourceWizardCustomization {

        public void CustomizeDataSourceWizard(IWizardCustomization<DashboardDataSourceModel> customization) {
            customization.StartPage = typeof(CustomChooseDataSourceNamePage<DashboardDataSourceModel>);
            customization.RegisterPage<CustomChooseDataSourceNamePage<DashboardDataSourceModel>, CustomChooseDataSourceNamePage<DashboardDataSourceModel>>();
        }
    }
    public class CustomChooseDataSourceNamePage<TModel> : ChooseDataSourceNamePage<DashboardDataSourceModel> where TModel : IDataSourceModel {
        public CustomChooseDataSourceNamePage(IChooseDataSourceNamePageView view, IDataSourceNameCreationService dataSourceNameCreator)
            : base(view, dataSourceNameCreator) {
        }
        public override Type GetNextPageType() {
            return typeof(DashboardChooseDataSourceTypePage<DashboardDataSourceModel>);
        }
    }
}

File: VB/WizardCustomizationNamePage/DataSourceWizardCustomization.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardCommon.DataSourceWizard
Imports DevExpress.DashboardWin.ServiceModel
Imports DevExpress.DataAccess.UI.Wizard
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Services
Imports DevExpress.DataAccess.Wizard.Views
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WizardCustomizationExample
    Public Class DataSourceWizardCustomization
        Implements IDashboardDataSourceWizardCustomization

        Public Sub CustomizeDataSourceWizard(ByVal customization As IWizardCustomization(Of DashboardDataSourceModel)) Implements IDashboardDataSourceWizardCustomization.CustomizeDataSourceWizard
            customization.StartPage = GetType(CustomChooseDataSourceNamePage(Of DashboardDataSourceModel))
            customization.RegisterPage(Of CustomChooseDataSourceNamePage(Of DashboardDataSourceModel), CustomChooseDataSourceNamePage(Of DashboardDataSourceModel))()
        End Sub
    End Class
    Public Class CustomChooseDataSourceNamePage(Of TModel As IDataSourceModel)
        Inherits ChooseDataSourceNamePage(Of DashboardDataSourceModel)

        Public Sub New(ByVal view As IChooseDataSourceNamePageView, ByVal dataSourceNameCreator As IDataSourceNameCreationService)
            MyBase.New(view, dataSourceNameCreator)
        End Sub
        Public Overrides Function GetNextPageType() As Type
            Return GetType(DashboardChooseDataSourceTypePage(Of DashboardDataSourceModel))
        End Function
    End Class
End Namespace
